src/Utilities/Compiler/Extensions/IMethodSymbolExtensions.cs

```diff
@@ -288,18 +288,30 @@ private static bool HasDisposeCloseAsyncMethodSignature(this IMethodSymbol metho
         }
 
         /// <summary>
-        /// Checks if the given method has the signature "override Task DisposeCoreAsync(bool)".
+        /// Checks if the given method has the signature "override Task DisposeCoreAsync(bool)" or "override Task DisposeAsyncCore(bool)".
         /// </summary>
         private static bool HasOverriddenDisposeCoreAsyncMethodSignature(this IMethodSymbol method, [NotNullWhen(returnValue: true)] INamedTypeSymbol? task)
         {
-            return method.Name == "DisposeCoreAsync" &&
+            return (method.Name == "DisposeCoreAsync" || method.Name == "DisposeAsyncCore") &&
```

Comment:
I would check the `DisposeAsyncCore` first, as it is the mostly used version
```suggestion
            return (method.Name == "DisposeAsyncCore" || method.Name == "DisposeCoreAsync") &&
```